### PR TITLE
Update to mirage 4.6.0

### PR DESCRIPTION
**Pull request by psafont**
The Console  module is not available anymore, as logging doesn't need it, and `xl console <domain>` still works to see the output.

Now returning a unit lwt.t is required, so change the return type of the unikernel

Lots of dependencies weren't needed so they got dropped.

### CODE REVIEW

File: src/config.ml [hunk 1]
```diff
@@ -1,22 +1,14 @@
 open Mirage
 
-let suspend =
```
Comment (edwintorok) [inline src/config.ml:3]: This is used later on when we look for it in the cmdline arguments.

Is there an equivalent with the new mirage version?

Comment (edwintorok) [reply src/config.ml:3]: Although I can't actually find where... maybe it was dead code?

Comment (psafont) [reply src/config.ml:3]: There's an equivalent to have runtime arguments, but i couldn't see where this argument was being used

**Comment by psafont** [comment]
Future work is enabling the lifecycle actions in the runtime, and change the lifecycle actions to perform a continuation (needed for suspend and delaying the actions)

**Comment by psafont** [comment]
I've got some more commit to submit. They change how the main loop works, removes commands that aren't easily triggered from xapi. It's still missing suspend functionality and advertising of supported shutdown features: https://github.com/xapi-project/xen-test-vm/compare/master...psafont:xen-test-vm:reorg

I'll open a PR when that's ready

**Pull request closed without merging**